lib/gias_api_client.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+require "zip"
+
+class GiasApiClient
+  def get_files
+    data = fetch_extract
+    temp_files_from_zipped_response(data)
+  end
+
+private
+
+  def fetch_extract
+    client = Savon.client(wsdl: Rails.application.config.gias_api_schema)
+    res = client.call(
+      :get_extract,
+      message: { "tns:Id" => Rails.application.config.gias_extract_id },
+      wsse_auth: [Rails.application.config.gias_api_user, Rails.application.config.gias_api_password],
+      wsse_timestamp: true,
+    )
+    parts = parse_body(res.http)
+    parts.second.body.to_s
```

Comment:
Is this how we get rid of the request headers?

Reply:
Yes, in essence. I adapted some code from a long dead library: [savon-multipart](https://github.com/savonrb/savon-multipart/blob/5e5a247f3fc5b5e2b25e2eaa76893fa34bda8adf/lib/savon/multipart/response.rb)